### DISCUSSION TITLE
Training SNN with random search process

### DISCUSSION THREAD

**Discussion by cnrgbot** [nengo/nengo-zoo#1298] [General Discussion]
> _Archived from the Nengo Discourse forum._ Originally posted by **vzanon** on 2024-01-12. (Forum topic #2731, post #1 / id 7658)

Hi guys, I'm training my neural network converted to SNN using nengo_dl. In my problem I need to obtain the best training parameters and so I decided to implement the random search process, since spiking networks do not have support for this... but I am getting the following result, is this normal to happen? What could be happening? Can anyone help me?

Image 01:
![image](https://raw.githubusercontent.com/nengo/forum-archive/main/uploads/default/original/2X/0/0cd99faef78f7503793db63d1f584d0fde875336.png)

Image 02:
![image](https://raw.githubusercontent.com/nengo/forum-archive/main/uploads/default/original/2X/5/51f234003638154492c9fb460f93eb99ba75a132.png)

It seems that there are times when the random search works (second image) and other times when it doesn't (first image)... does anyone know or has anyone experienced this? 

Thank you.

**Comment by cnrgbot** [comment]
> _Archived from the Nengo Discourse forum._ Originally posted by **vzanon** on 2024-01-26. (Forum topic #2731, post #2 / id 7676)

I solved this problem by inserting a Batch Normalization layer into my SNN structure. If you have a problem with network convergence, insert this layer as it helps to minimize the effects of erroneously calculated error gradients across the layers.
